File: app/src/main/java/com/commit451/gitlab/viewHolder/FeedEntryViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.commit451.gitlab.App;
import com.commit451.gitlab.R;
import com.commit451.gitlab.model.rss.Entry;
import com.commit451.gitlab.transformation.CircleTransformation;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Represents the view of an item in the RSS feed
 */
public class FeedEntryViewHolder extends RecyclerView.ViewHolder {

    public static FeedEntryViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_entry, parent, false);
        return new FeedEntryViewHolder(view);
    }

    @BindView(R.id.image)
    ImageView image;
    @BindView(R.id.title)
    TextView textTitle;
    @BindView(R.id.description)
    TextView textSummary;

    public FeedEntryViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Entry entry) {
        App.get().getPicasso()
                .load(entry.getThumbnail().getUrl())
                .transform(new CircleTransformation())
                .into(image);

        textTitle.setText(Html.fromHtml(entry.getTitle()));
        textSummary.setText(Html.fromHtml(entry.getSummary()));
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/FeedEntryViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import coil.api.load
import coil.transform.CircleCropTransformation
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.extension.formatAsHtml
import com.commit451.gitlab.model.rss.Entry
import com.commit451.gitlab.util.DateUtil

/**
 * Represents the view of an item in the RSS feed
 */
class FeedEntryViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): FeedEntryViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_entry, parent, false)
            return FeedEntryViewHolder(view)
        }
    }

    private val image: ImageView by bindView(R.id.image)
    private val textTitle: TextView by bindView(R.id.title)
    private val textSummary: TextView by bindView(R.id.description)
    private val textUpdated: TextView by bindView(R.id.updated)

    fun bind(entry: Entry) {
        image.load(entry.thumbnail.url) {
            transformations(CircleCropTransformation())
        }

        textTitle.text = entry.title.formatAsHtml()
        textSummary.text = entry.summary.formatAsHtml()
        val updatedRelative = DateUtil.getRelativeTimeSpanString(itemView.context, entry.updated)

        textUpdated.text = updatedRelative
    }
}
